ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/committer/PackageCommitter.scala

```diff
@@ -200,19 +201,22 @@ final private[kvutils] class PackageCommitter(
   }
 
   private def decodePackages(
-      archives: Iterable[DamlLf.Archive]
+      archives: Iterable[Raw.Archive]
   ): Either[String, Map[Ref.PackageId, Ast.Package]] =
     metrics.daml.kvutils.committer.packageUpload.decodeTimer.time { () =>
       type Result = Either[List[String], Map[Ref.PackageId, Ast.Package]]
       archives
-        .foldLeft[Result](Right(Map.empty)) { (acc, arch) =>
+        .foldLeft[Result](Right(Map.empty)) { (acc, rawArchive) =>
           try {
-            acc.map(_ + lf.archive.Decode.assertDecodeArchive(arch))
+            acc.map { result =>
+              val archive = ArchiveParser.assertFromByteString(rawArchive.bytes)
```

Comment:
I think this is a valuable remark. Based on my discussion with @cocreature and @remyhaemmerle-da, parsing should be cheap. I'm not sure if we want to optimize for that before even knowing if the impact is significant.

Reply:
It's not just an optimization for the computer, it also de-duplicates code, which is an optimization for my brain when reading it. :wink:

Reply:
Ok, let me reshuffle this a bit 😉

Reply:
I did what I could. There are still 2 functions that parse the archives. These 2 functions will most likely go to `kv-support`. I don't want to store the archives to avoid Daml-LF references.